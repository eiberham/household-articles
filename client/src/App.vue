<template>
    <div id="app">
        <Header />
        <Sidebar @selected="onSelected" />
        <div class="content" v-if="products">
            <Product :title="title" :products="products" />
        </div>
    </div>
</template>

<script>
import axios from 'axios';

import Header from './components/Header.vue';
import Sidebar from './components/Sidebar.vue';

export default {
    components: {
        Header,
        Sidebar,
        Product: () => import(/* webpackChunkName: "product" */ './components/Product')
    },
    data(){
        return {
            title: '',
            products: [],
        }
    },
    methods: {
        onSelected: function(_type){
            this.title = _type
            this.makeRequest(_type);
        },
        makeRequest: async function(type) {
            try {
                const response = await axios.get(`http://localhost:3000/api/${type}`, {}, {
                    headers: {
                        'content-type': 'application/json',
                    }
                });
                
                this.products = response.data;
            } catch (_err) {
                console.error(_err);
            }
        }
    }
}
</script>

<style lang="scss">
    body {
        margin: 0;
        padding: 0;
    }

    #app {
        box-sizing: border-box;
        display: grid;
        grid: {
            template-columns: 100px 1fr;
            template-rows: 100px 1fr;
            template-areas: "header header"
                            "sidebar content";
            gap: 0;
        }
        min: {
            height: 100vh;
            width: 100%;
        }
        margin: 0;
        padding: 0;
    }
</style>