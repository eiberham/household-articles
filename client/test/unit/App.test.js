import { mount } from "@vue/test-utils";
import App from "../../src/App.vue";

describe("App", () => {
  const wrapper = mount(App, {});

  test("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test("matches snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
