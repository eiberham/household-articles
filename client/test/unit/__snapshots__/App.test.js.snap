// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  id="app"
>
  <section
    class="header"
  >
    <select>
      <option>
        Guardar y salir
      </option>
       
      <option>
        Salir sin guardar
      </option>
       
      <option>
        Guardar y continuar
      </option>
    </select>
  </section>
   
  <section
    class="sidebar"
  >
    <nav
      class="menu"
    >
      <ul
        class="menu__wrapper"
        role="list"
      >
        <li
          class="menu__item opening"
          data-name="opening"
        >
          
        Opening
      
        </li>
         
        <li
          class="menu__item equipment"
          data-name="equipment"
        >
          
        Equipment
      
        </li>
         
        <li
          class="menu__item ending"
          data-name="ending"
        >
          
        Ending
      
        </li>
      </ul>
    </nav>
  </section>
   
  <div
    class="content"
  >
    <section
      class="opening"
    >
      <div
        class="product-list hide"
      >
        <div
          class="product-list__toggle"
          role="button"
        >
          <i
            class="toggle-icon"
          />
        </div>
         
        <h1
          class="product-list__title"
        >
          
      
    
        </h1>
         
      </div>
       
      <div
        class="actions"
      >
        <div
          class="config"
        >
          <button
            class="button"
            type="button"
          >
            
        Fijar
      
          </button>
           
          <button
            class="button"
            type="button"
          >
            
        Borrar
      
          </button>
        </div>
         
        <div
          class="control"
        >
          <div
            class="control-zoom"
          >
            <button
              class="button plus"
              type="button"
            />
             
            <button
              class="button minus"
              type="button"
            />
          </div>
           
          <div
            class="control-move"
          >
            <button
              class="button move"
              type="button"
            />
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
